service/src/main/java/uk/nhs/adaptors/gp2gp/TransformJsonToXml.java

```diff
@@ -0,0 +1,225 @@
+import uk.nhs.adaptors.gp2gp.ehr.mapper.OutputMessageWrapperMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ParticipantMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.PertinentInformationObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.PractitionerAgentPersonMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.RequestStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.StructuredObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.DiagnosticReportMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.ObservationMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.SpecimenMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class TransformJsonToXml {
+
+    private static final String JSON_FILE_INPUT_PATH =
+            Paths.get("src").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/input/";
+    private static final String XML_OUTPUT_PATH =
+            Paths.get("src").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/output/";
+    private static final Logger LOGGER = Logger.getLogger(TransformJsonToXml.class.getName());
+
+    private static final FhirParseService FHIR_PARSE_SERVICE = new FhirParseService();
+
+    public static void main(String[] args) throws Exception {
+
+        var inputWrapper = getFiles();
+
+        for (int i = 0; i < inputWrapper.getJsonFileInputs().size(); i++) {
+            var jsonString = inputWrapper.getJsonFileInputs().get(i);
+            String xmlResult = mapJsonToXml(jsonString);
+            var fileName = inputWrapper.getJsonFileNames().get(i);
+            writeToFile(xmlResult, fileName);
+        }
+    }
+
+    private static InputWrapper getFiles() {
+
+        File[] files = new File(JSON_FILE_INPUT_PATH).listFiles();
+        List<String> jsonStringInputs = new ArrayList<>();
+        List<String> fileNames = new ArrayList<>();
+
+        for (File inputJsonFile : files) {
+            LOGGER.info("Parsing File: " + inputJsonFile.getName());
+            if (FilenameUtils.getExtension(inputJsonFile.getName()).equalsIgnoreCase("json")) {
+                try {
+                    String jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + inputJsonFile.getName());
+                    jsonStringInputs.add(jsonAsString);
+                    fileNames.add(inputJsonFile.getName());
+                } catch (Exception e) {
+                    LOGGER.info("Could Not Read Json Files.");
+                }
+            } else {
+                LOGGER.info("No .json Files Found");
+            }
+        }
```

Comment:
This message is a little misleading - it will be printed every time a file is found that doesn't have a json extension. So if you had a directory of 2 .json files and 3 .xml files, it would print that not json files were found, 3 times!

I suggest detecting if the array is empty before looping. You might also consider that to be an exception scenario.

Then you can loop with more idiomatic Java 8+, especially if you add a lombok `@SneakyThrows` on `readJsonFileAsString()`

```suggestion
        if (files.length == 0) {
            throw new Exception("No json files found");
        }
        Arrays.stream(files)
            .peek(file -> LOGGER.info("Parsing file: {}", file.getName()))
            .filter(file -> FilenameUtils.getExtension(file.getName()).equalsIgnoreCase("json"))
            .forEach(file -> {
                String jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + inputJsonFile.getName());
                jsonStringInputs.add(jsonAsString);
                fileNames.add(inputJsonFile.getName());
            });
```